fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PartitionPruneRule.java

```diff
@@ -66,12 +70,27 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
                     .filter(id -> table.getPartition(id).hasData()).collect(Collectors.toList());
         }
 
+        if (isNeedFurtherPrune(selectedPartitionIds, olapScanOperator, partitionInfo)) {
```

Comment:
Needn't check the predicates format here?  do we need to double prune in some cases?